examples/text/amazonreviewpolarity.py

```diff
@@ -37,25 +32,29 @@ def AmazonReviewPolarity(root, split):
     """Demonstrating caching, extraction and sanity check pipelines."""
 
     url_dp = IterableWrapper([URL])
-    # cache data on-disk with sanity check
-    cache_dp = url_dp.on_disk_cache(
+    cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
-    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
-
-    cache_dp = FileOpener(cache_dp, mode="b")
-
-    # stack TAR extractor on top of loader DP
-    extracted_files = cache_dp.read_from_tar()
-
-    # filter files as necessary
-    filter_extracted_files = extracted_files.filter(lambda x: split in x[0])
-
-    # stack sanity checker on top of extracted files
-    check_filter_extracted_files = filter_extracted_files.check_hash(
-        {os.path.normpath(os.path.join(root, _EXTRACTED_FILES[split])): _EXTRACTED_FILES_MD5[split]},
-        "md5",
-    )
-
-    # stack CSV reader and do some mapping
-    return check_filter_extracted_files.parse_csv().map(fn=lambda t: (int(t[0]), t[1]))
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    def extracted_filepath_fn(x):
```

Comment:
@ejguan I am not sure why is it necessary to pass all the files to generator as per your comment here (https://github.com/pytorch/data/pull/169#discussion_r787049096). I guess it is sufficient to only check for file(s) we are concerned with, right? not sure if I miss anything here?

Reply:
The second `on_disk_cache` is used to cache the decompressed files from an archive. So, for `on_disk_cache`, it only gets the input archive file path as `xxx.tar.gz` but we want to check if all decompressed files existing on your file system. This is a 1-to-N operation. So, in order to check all of the decompressed files, we have to accept a generator function as the `filepath_fn`.

Reply:
I see, got it. Ya, In this case, I am only need to check the existence of single file (even though there are more in the archive). I am guess it would be OK to specify only files we are concerned with and filter out remaining ones in the pipe?